libs/s25main/world/MapBase.cpp

```diff
@@ -138,6 +138,11 @@ unsigned MapBase::CalcDistance(const Position& p1, const Position& p2) const
     return ((dy + (dx > 0 ? dx : 0)) / 2);
 }
 
+unsigned MapBase::CalcMaxDistance() const
+{
+    return CalcDistance(MapPoint::all(0), MapPoint(size_.x / 2, size_.y / 2));
```

Comment:
Out of curiosity I resolved this to `return (size.x > size.y/2) ? size.x/2 + size.y/4 : size.y/2` which isn't intuitive but I'm pretty sure all is correct (i.e. `CalcDistance`), so leave this as-is
However I omitted one part: If size.y is not a multiple of 4 then the maxdistance is one less.
Got to check that later...

Reply:
The settler coordinate system gave me quite a headache in the past. But yeah, your solution should work as well. Since it's not performance critical at all, I guess it's just about what's more intuitive. On the other hand, I think as a dev I'd just  expect those functions (like `GetNeighbor`, `CalcDistance`, etc.) just to work without looking into the code if not absolutely necessary. So I wouldn't mind changing it, but either way is fine ...

Reply:
Fully agreed, the hexagonal map is difficult... But I don't see how to make it easier to work with so it is what it is.
No, don't change, as mentioned there is the corner case of non-multiple-of-4 which I omitted and I'd say we would need a test for this function if we change it to that (iterate over all points and check that no distance is bigger than this. but that would be quite expensive...)